subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc

```diff
@@ -196,3 +196,20 @@ Gradle used to have two ways of publishing artifacts.
 Now, the situation has been cleared and all build should use the `maven-publish` plugin.
 The last remaining artifact of the old way of publishing is the `Upload` task that has been deprecated and scheduled for removal in Gradle 8.0.
 Existing clients should migrate to the <<publishing_maven.adoc#publishing_maven,`maven-publish` plugin>>.
+
+[[base_convention_deprecation]]
+=== Deprecated `base` plugin conventions
+
+The convention properties contributed by the `base` plugin have been deprecated and scheduled for removal in Gradle 8.0.
+The conventions are replaced by the the `base { }` configuration block backed by link:{groovyDslPath}/org.gradle.api.plugins.BasePluginExtension.html[BasePluginExtension].
+
+The old convention object defines the `distsDirName`, `libsDirName` and `archivesBaseName` properties with simple getter and setter methods.
+For compatibility reasons, those are still available in the extension but only for compatibility purposes.
+Build scripts should solely use the properties of type `Property`:
+```
+base {
+    archivesName = 'customBase'
+    distsDirectory = layout.buildDirectory.dir('custom-dist')
+    libsDirectory = layout.buildDirectory.dir('custom-libs')
+}
+```
```

Comment:
Is it possible to link to a snippet from the upgrade guide, like it is done from base_plugin.adoc?

Reply:
I don't think we want that.

In this case, we want to capture how to upgrade from Gradle 7.0 to 7.1. If we link the snippet to the upgrade guide and later - say, in 7.2 - change the snippet, the freshly rendered upgrade guide might have instructions that's invalid for the 7.0 -> 7.1 upgrade.